hangman/lib/game/game.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Hangman.Game do
+  @moduledoc """
+  Implemnation of Hangman API.
+  """
+
+  #file with helper methods
+  import Hangman.Assist
+
+  #state structure
+  defstruct turns_left: 7,
+            game_state: :initializing,
+            letters: [],
+            used: [],
+            word: "",
+            last_guess: ""
+  
+  #new_game() method that takes a random word and uses it  
+  def new_game() do
+    word = Dictionary.random_word()
+    %Hangman.Game{}
+    |> Map.put(:word, word)
+    |> Map.put(:letters, List.duplicate("-", String.length(word)))
+  end
+
+  #tally method that returns state without word
+  def tally(game), do: Map.from_struct(game) |> Map.delete(:word)
+
+  #main function of the API, does all the movement and work
+  def make_move(game, move) do
```

Comment:
## -2 idiomatic

I see what you're doing, but it's a little complex: the `guess` function ends up taking 6 parameters, which are hard to keep track of and reason about.

I'd rather see you structure this as a decision tree, splitting it into maybe three or four functions, each a line or two long.